diff --git a/addons/account/account.py b/addons/account/account.py
index 2bae5c5..8242305 100644
--- a/addons/account/account.py
+++ b/addons/account/account.py
@@ -2114,7 +2114,7 @@ class account_tax(osv.osv):
         tax_compute_precision = precision
         if taxes and taxes[0].company_id.tax_calculation_rounding_method == 'round_globally':
             tax_compute_precision += 5
-        totalin = totalex = round(price_unit * quantity, precision)
+        totalin = totalex = round(price_unit * quantity, tax_compute_precision)
         tin = []
         tex = []
         for tax in taxes:
